walletconnect-wallet-standard/src/index.js

```diff
@@ -73,265 +63,477 @@ const getConnectParams = (chains) => ({
+  /** @returns {WalletWithSolanaFeatures["chains"]} */
+  get chains() {
+    return this.#chains;
+  }
 
-  chains = [WalletConnectChainID.Mainnet, WalletConnectChainID.Devnet];
+  /** @returns {WalletWithSolanaFeatures["accounts"]} */
+  get accounts() {
+    const all = Object.values(this.#accounts).reduce((arr, chainIdSubArr) => [
+      ...arr,
+      ...chainIdSubArr,
+    ]);
+    return all.map(this.#walletConnectAccountToStandardAccount.bind(this));
+  }
 
-  /**
-   * @returns {WalletConnectClient}
-   */
-  get client() {
-    if (this._client) {
-      return this._client;
+  /** @returns {WalletWithSolanaFeatures["features"] & import("@wallet-standard/features").StandardConnectFeature & import("@wallet-standard/features").StandardDisconnectFeature & import("@wallet-standard/features").StandardEventsFeature} */
+  get features() {
+    return {
+      // TODO: signMessage currently only supports ascii messages
+      "solana:signMessage": {
+        version: "1.0.0",
+        signMessage: this.#signMessage.bind(this),
+      },
+      "solana:signTransaction": {
+        version: "1.0.0",
+        supportedTransactionVersions: this.supportedTransactionVersions,
+        signTransaction: this.#signTransaction.bind(this),
+      },
+      "standard:connect": {
+        version: "1.0.0",
+        connect: this.#connect.bind(this),
+      },
+      "standard:disconnect": {
+        version: "1.0.0",
+        disconnect: this.#disconnect.bind(this),
+      },
+      "standard:events": {
+        version: "1.0.0",
+        on: this.#on.bind(this),
+      },
+    };
+  }
+
+  // TODO: implement querying the WalletConnect for its supported tx versions
+  /** @returns {import("@solana/wallet-standard-features").SolanaSignTransactionFeature["solana:signTransaction"]["supportedTransactionVersions"]} */
+  get supportedTransactionVersions() {
+    return ["legacy", 0];
+  }
+
+  /** @returns {NonLocalnetChain} */
+  get defaultChain() {
+    const res = this.#chains[0];
+    if (!res) {
+      throw new NoChainsSetError();
     }
-    throw new ClientNotInitializedError();
+    return res;
   }
 
+  /** @type {WalletConnectClient | undefined} */
+  #client;
+
+  /** @type {WalletConnectModal | undefined} */
+  #modal;
+
+  /** @type {import('@walletconnect/types').SignClientTypes.Options} */
```

Comment:
```suggestion
  /** @type {import('@walletconnect/types').SignClientTypes.Options  & { projectId: string }} */
```

to fix type error `string | undefined` in `new WalletConnectModal()`. It's marked as required in `SolanaWalletConnectWalletStandardCtorArgs` so this should be safe.